### PR TITLE
Add `dist: xenial` to linux Python 3.7 in travis.yml

### PR DESCRIPTION
**Pull request by janash**
addresses #62

### CODE REVIEW

**Comment by dgasmith** [comment]
Are you still getting this error? It would be good if we could leave the container unpinned if possible.

**Comment by j-wags** [comment]
I had to pin this last week when we were making the propertyestimator repo.

**Comment by dgasmith** [comment]
I guess they do not change it very quickly so it shouldn't be too much of a problem. The heart of this however is that we do not actually need the `python: 3.6` statements as we are using Conda and that statement only effects base python.

@Lnaden What if we pull this in and then make a new issue that pulls out the  `python: 3.6` statements for conda builds, and removes the `python env` stuff for pip builds?

@janash Could you do this by chance?

**Comment by Lnaden** [comment]
I think the issue is that if we pin the dist now, we will have to unpin later, and I'm not sure nuances of Travis CI are what the cookiecutter should deal with. For now, Travis uses Trusty as their default (which does not have Py37), and I don't know how long until they move to Xenial for their defaults. Could be a while. 

> we do not actually need the `python: 3.6`

We do for the Linux Pip installs, the Conda ones don't though.

> and removes the `python env` stuff for pip builds?

The python env things only exist for the OSX builds, which the Trusty/Xenial issues won't effect anyways. However, Python is not officially supported by Travis's OSX builds, so if you want to use `pip` on Travis + OSX, then we needed a reliable python install, so I used Brew + pyenv to get it all.

tl;dr. Not sure what you are asking me to do unless we drop pip support, which I dont think we want

**Comment by dgasmith** [comment]
Switch to language generic:
```
language: generic

matrix:
  include:
    # Extra includes for OSX since python language is not available by default on OSX
    - os: osx
      env: PYTHON_VER=3.6

    - os: linux
      python: 3.7 # If using linux pip
      env: PYTHON_VER=3.7 # If using conda, both are not needed.
```

We have unneeded information in the matrix build. This will allow everything to work without a pinned container except for `Pip + Linux + Python 3.7` now.

**Comment by Lnaden** [comment]
Ah, so you are suggesting minimizing the number of pins in the build matrix as a whole, and only pinning `xenial` where we absolutely have to?

**Comment by dgasmith** [comment]
Yea, would help I think as brew/conda do not require any image base. We did this inside QCFractal to let 3.7 builds though without pinning containers.

It looks like Xenial came out in September, 2018. Might be a bit before they default it (~6-12 months).

**Comment by Lnaden** [comment]
> We did this inside QCFractal to let 3.7 builds though without pinning containers.

Kind of. We made `language` generic since we are only using conda, but we still pinned to the Xenial `dist`

https://github.com/MolSSI/QCFractal/blob/master/.travis.yml#L7-L10

**Comment by dgasmith** [comment]
Whoops, looks like I forgot to unpin there.
